web/ui/react-app/src/pages/Services.tsx

```diff
@@ -1,15 +1,116 @@
 import React, { FC } from 'react';
 import { RouteComponentProps } from '@reach/router';
 import PathPrefixProps from '../PathPrefixProps';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faSpinner } from '@fortawesome/free-solid-svg-icons';
 import { Alert } from 'reactstrap';
+import { useFetch } from '../utils/useFetch';
+import { LabelsTable } from './LabelsTable';
+import { Target, Labels, DroppedTargets } from './targets/target';
 
-const Services: FC<RouteComponentProps & PathPrefixProps> = ({ pathPrefix }) => (
-  <>
-    <h2>Service Discovery</h2>
-    <Alert color="warning">
-      This page is still under construction. Please try it in the <a href={`${pathPrefix}/service-discovery`}>Classic UI</a>.
-    </Alert>
-  </>
-);
+// TODO: Deduplicate with https://github.com/prometheus/prometheus/blob/213a8fe89a7308e73f22888a963cbf9375217cd6/web/ui/react-app/src/pages/targets/ScrapePoolList.tsx#L11-L14
+interface ServiceMap {
+  activeTargets: Target[];
+  droppedTargets: any[];
+}
+
+export interface TargetLabels {
+  discoveredLabels: Labels;
+  labels: Labels;
+}
+
+const Services: FC<RouteComponentProps & PathPrefixProps> = ({ pathPrefix }) => {
+  const { response, error } = useFetch<ServiceMap>(`${pathPrefix}/api/v1/targets`);
+
+  const processSummary = (response: ServiceMap) => {
+    const targets: any = {};
+
+    // Get targets of each type along with the total and active end points
+    for (const target of response.activeTargets) {
+      const { scrapePool: name } = target;
+      if (!targets[name]) {
+        targets[name] = {
+          total: 0,
+          active: 0,
+        };
+      }
+      targets[name].total++;
+      targets[name].active++;
+    }
+    for (const target of response.droppedTargets) {
+      const { job: name } = target.discoveredLabels;
+      if (!targets[name]) {
+        targets[name] = {
+          total: 0,
+          active: 0,
+        };
+      }
+      targets[name].total++;
+    }
+
+    return targets;
+  };
+
+  const processTargets = (response: Target[], dropped: DroppedTargets[]) => {
+    const labels: Record<string, TargetLabels[]> = {};
+
+    for (const target of response) {
+      const name = target.scrapePool;
+      if (!labels[name]) {
+        labels[name] = [];
+      }
+      labels[name].push({
+        discoveredLabels: target.discoveredLabels,
+        labels: target.labels,
+      });
+    }
+
+    for (const target of dropped) {
+      const { job: name } = target.discoveredLabels;
+      if (!labels[name]) {
+        labels[name] = [];
+      }
+      labels[name].push({
+        discoveredLabels: target.discoveredLabels,
+        labels: { dropped: 'yes' },
```

Comment:
A normal (non-dropped) target may include a label with the label name `dropped`, so you cannot just use labels to communicate whether a target has been dropped. You'll probably want to add a `dropped` boolean or something outside of the labels.

Reply:
Sure. I have implemented an external boolean property.